libraries/chain/contracts/abi_serializer.cpp

```diff
@@ -48,6 +48,10 @@ namespace eosio { namespace chain { namespace contracts {
       );
    }
 
+   struct symbol_name {
+      uint64_t value;
+   };
```

Comment:
Should `symbol_name` be a `name` and be defined in types.hpp along with action_name, scope_name, account_name, etc?

Reply:
`symbol_name` cannot be a `name` because it must consist of uppercase letters. Perhaps the `_name` suffix is misleading and it should be renamed? But I can't think of a better name right now.

I think it would be more appropriate to put `symbol_name` into symbol.hpp rather than types.hpp. Expect that commit momentarily.

Reply:
Yes, symbol.hpp for it and the variant code would be better. How about `symbol_code` I know code is very overloaded, but here at least those things are called currency codes. http://www.xe.com/symbols.php

I have also seen them called "Currency Abbreviation".

Reply:
I like `symbol_code`. 

Also, I think the names and methods of the `symbol`-related classes in contracts/eosiolib/asset.hpp should be changed to mirror the symbol.hpp on the native side (and perhaps vice versa), but that is a task for another time when I do the token refactor work.